models/Document.php

```diff
@@ -879,6 +879,7 @@ public function getFullPath(bool $force = false)
         if (!$link && \Pimcore\Tool::isFrontend() && Site::isSiteRequest() && !FrontendTool::isDocumentInCurrentSite($this)) {
             if ($masterRequest && ($masterDocument = $masterRequest->get(DynamicRouter::CONTENT_KEY))) {
                 if ($masterDocument instanceof WrapperInterface) {
+                    $hardlinkPath = null;
```

Comment:
```suggestion
                    $hardlinkPath = '';
```
Better use a `string` here, as `strpos()` would convert it otherwise.